Unit.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+#include <string>
+class Unit
+{
+private:
+	int hp;
+	int dmg;
+	std::string name;
+public:
+	void SetHp(int HP);

```

Comment:
ez egy tul altalanos public fuggveny, ilyet nem akarunk kitenni, hogy akarki akarmennyire beallithassa.

Reply:
A setHp függvényt teljes egészében elimináltuk. Helyette egy takeDamage függvényt vezettünk be, amivel csak meghatározott mértékű sebzést lehet bevinni, de konkrétan nem lehet megadni a hp értékét.